service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/StructuredObservationValueMapper.java

```diff
@@ -40,6 +46,31 @@ public String mapObservationValueToStructuredElement(IBaseElement value) {
             .apply(value);
     }
 
+    public String mapInterpretation(CodeableConcept interpretation) {
+        Coding coding = interpretation.getCodingFirstRep();
+        String code = coding.getCode();
+
+        switch (code) {
+            case "H":
+            case "HH":
+            case "HU":
+                return String.format(INTERPRETATION_CODE_TEMPLATE, "HI", "2.16.840.1.113883.2.1.6.5",
+                    "Above high reference limit", coding.getDisplay());
+            case "L":
+            case "LL":
+            case "LU":
+                return String.format(INTERPRETATION_CODE_TEMPLATE, "LO", "2.16.840.1.113883.2.1.6.5",
+                    "Below low reference limit", coding.getDisplay());
+            case "A":
+            case "AA":
+                return String.format(INTERPRETATION_CODE_TEMPLATE, "PA", "2.16.840.1.113883.2.1.6.5",
+                    "Potentially abnormal", coding.getDisplay());
+            default:
+                return StringUtils.EMPTY;
+        }
```

Comment:
FYI: from JDK 12 you'll be able to write this with a switch expression

```suggestion
        return switch (code) {
            case "H", "HH", "HU" -> String.format(INTERPRETATION_CODE_TEMPLATE, "HI", "2.16.840.1.113883.2.1.6.5",
                    "Above high reference limit", coding.getDisplay());
            case "L", "LL", "LU" -> String.format(INTERPRETATION_CODE_TEMPLATE, "LO", "2.16.840.1.113883.2.1.6.5",
                    "Below low reference limit", coding.getDisplay());
            case "A", "AA" -> String.format(INTERPRETATION_CODE_TEMPLATE, "PA", "2.16.840.1.113883.2.1.6.5",
                    "Potentially abnormal", coding.getDisplay());
            default -> StringUtils.EMPTY;
        }
```